File: 0125-valid-palindrome/0125-valid-palindrome.cpp
class Solution {
public:
    bool isPalindrome(string s) {
        int n=s.length();
        string t="";
        for(int i=0;i<n;i++){
            char c=s[i];
            if(c>='A'&&c<='Z'){
                c=c-'A'+'a';
            }
            if((c>='a'&&c<='z')||(c>='0'&&c<='9')){
                t+=c;
            }
        }
        int l=0;
        int r=t.length()-1;
        while(l<r){
            if(t[l]!=t[r]){
                return 0;
            }
            l++;
            r--;
        }
        return 1;
    }
};

File: 0125-valid-palindrome/0125-valid-palindrome.py
class Solution:
    def isPalindrome(self, s: str) -> bool:
        s=s.lower()
        n=len(s)
        # applying the 2 pointers logic
        i=0 # starts from right
        j=n-1 #starts from the left
        while i<j:          
            if not s[i].isalnum():  #check if not in left
                i+=1      
                continue
            if not s[j].isalnum():  #when not in right
                j-=1         
                continue
            if s[i]!=s[j]:   #check if not a pal (left !=right)   
                return False         
            i+=1                   
            j-=1
        return True       